src/services/codefixes/convertFunctionToEs6Class.ts

```diff
@@ -102,51 +141,76 @@ namespace ts.codefix {
-                            /*typeParameters*/ undefined, functionExpression.parameters, /*type*/ undefined, functionExpression.body);
-                        copyLeadingComments(assignmentBinaryExpression, method, sourceFile);
-                        return method;
-                    }
-
-                    case SyntaxKind.ArrowFunction: {
-                        const arrowFunction = assignmentBinaryExpression.right as ArrowFunction;
-                        const arrowFunctionBody = arrowFunction.body;
-                        let bodyBlock: Block;
-
-                        // case 1: () => { return [1,2,3] }
-                        if (arrowFunctionBody.kind === SyntaxKind.Block) {
-                            bodyBlock = arrowFunctionBody as Block;
-                        }
-                        // case 2: () => [1,2,3]
-                        else {
-                            bodyBlock = createBlock([createReturn(arrowFunctionBody)]);
-                        }
-                        const fullModifiers = concatenate(modifiers, getModifierKindFromSource(arrowFunction, SyntaxKind.AsyncKeyword));
-                        const method = createMethod(/*decorators*/ undefined, fullModifiers, /*asteriskToken*/ undefined, memberDeclaration.name, /*questionToken*/ undefined,
-                            /*typeParameters*/ undefined, arrowFunction.parameters, /*type*/ undefined, bodyBlock);
-                        copyLeadingComments(assignmentBinaryExpression, method, sourceFile);
-                        return method;
+                    case SyntaxKind.FunctionExpression:
+                    case SyntaxKind.ArrowFunction:
+                        return createFunctionLikeExpressionMember(members, <FunctionExpression | ArrowFunction>assignmentBinaryExpression.right, (<PropertyAccessExpression>memberDeclaration).name);
+                    case SyntaxKind.ObjectLiteralExpression: {
+                        return map(
+                            (<ObjectLiteralExpression>assignmentBinaryExpression.right).properties,
+                            property => {
+                                if (isMethodDeclaration(property) || isGetOrSetAccessorDeclaration(property)) {
+                                    // MethodDeclaration and AccessorDeclaration can appear in a class directly
+                                    return members.concat(property);
+                                }
+                                // Drop constructor assignments
+                                if (isConstructorAssignment(property)) return members;
+                                return createFunctionLikeExpressionMember(members, <FunctionExpression | ArrowFunction>(<PropertyAssignment>property).initializer, property.name!);
+                            }
+                        ).reduce((x, y) => x.concat(y));

```

Comment:
Nit: just use `flatMap` instead of `map` + reduce

Reply:
Oh, I didn't notice that. that's the original code, will change soon